dynamic_env_pytest_tests/tests/services/s3_gate/test_s3_gate.py

```diff
@@ -0,0 +1,559 @@
+import logging
+import os
+from random import choice, choices
+
+import allure
+import pytest
+from aws_cli_client import AwsCliClient
+from common import ASSETS_DIR
+from file_helper import (
+    generate_file,
+    generate_file_with_content,
+    get_file_content,
+    get_file_hash,
+    split_file,
+)
+from s3_helper import (
+    check_objects_in_bucket,
+    check_tags_by_bucket,
+    check_tags_by_object,
+    set_bucket_versioning,
+    try_to_get_objects_and_expect_error,
+    parametrize_clients
+)
+
+from pytest_tests.steps import s3_gate_bucket, s3_gate_object
+from s3.s3_gate_base import TestNeofsS3GateBase
+
+logger = logging.getLogger("NeoLogger")
+
+
+def pytest_generate_tests(metafunc):
+    parametrize_clients(metafunc)
+
+
+@allure.link("https://github.com/nspcc-dev/neofs-s3-gw#neofs-s3-gateway", name="neofs-s3-gateway")
+@pytest.mark.s3_gate
+@pytest.mark.s3_gate_base
+class TestS3Gate(TestNeofsS3GateBase):
+    @allure.title("Test S3 Bucket API")
+    def test_s3_buckets(self, simple_object_size):
+        """
+        Test base S3 Bucket API (Create/List/Head/Delete).
+        """
+
+        file_path = generate_file(simple_object_size)
+        file_name = self.object_key_from_file_path(file_path)
+
+        with allure.step("Create buckets"):
+            bucket_1 = s3_gate_bucket.create_bucket_s3(self.s3_client, True, bucket_configuration="rep-1")
+            set_bucket_versioning(self.s3_client, bucket_1, s3_gate_bucket.VersioningStatus.ENABLED)
+            bucket_2 = s3_gate_bucket.create_bucket_s3(self.s3_client, bucket_configuration="rep-1")
+
+        with allure.step("Check buckets are presented in the system"):
+            buckets = s3_gate_bucket.list_buckets_s3(self.s3_client)
+            assert bucket_1 in buckets, f"Expected bucket {bucket_1} is in the list"
+            assert bucket_2 in buckets, f"Expected bucket {bucket_2} is in the list"
+
+        with allure.step("Bucket must be empty"):
+            for bucket in (bucket_1, bucket_2):
+                objects_list = s3_gate_object.list_objects_s3(self.s3_client, bucket)
+                assert not objects_list, f"Expected empty bucket, got {objects_list}"
+
+        with allure.step("Check buckets are visible with S3 head command"):
+            s3_gate_bucket.head_bucket(self.s3_client, bucket_1)
```

Comment:
im newbee in Python functionality. IIUC if this instruction throws any exception, the test step will fail, right?

Reply:
> im newbee in Python functionality. IIUC if this instruction throws any exception, the test step will fail, right?

Yes, it will